v2python/downgrader.py

```diff
@@ -0,0 +1,49 @@
+# Copyright © 2023-2024 Advanced Micro Devices, Inc.
+# SPDX-License-Identifier: MIT
+
+'''
+Used in conjunction with PARTIALLY_TUNED_FUNCTIONALS
+
+Commonly enabling functionals will cost extra resources,
+and thus make the fallback turing information unusable
```

Comment:
Could you elaborate on what this enables?  Was this the answer to tuning regressions?

Reply:
This allows `KernelDescription` objects to "downgrade"/"patch" some tuning parameters according to enabled functionals.

This function is not used in current builds, and was introduced because Triton/tritonsrc combinations, the maximal `(BLOCK_M,BLOCK_N)` is `32,16` when dropout becomes true, and at that time the tuning parameters has to be specified by `KernelDescription` class.

Now the selection process is largely handled by the tuning database, but the `DOWNGRADER` member is still supported by `KernelDescription` class for possible future use.